js/apps/admin-ui/src/dashboard/Dashboard.tsx

```diff
@@ -67,48 +69,46 @@ const EmptyDashboard = () => {
   );
 };
 
+const FeatureItem = ({ feature }: { feature: FeatureRepresentation }) => {
+  const { t } = useTranslation("dashboard");
+  return (
+    <ListItem key={feature.name} className="pf-u-mb-sm">
+      {feature.name}{" "}
+      {feature.type === FeatureType.EXPERIMENTAL ? (
+        <Label color="orange">{t("experimental")}</Label>
+      ) : null}
+      {feature.type === FeatureType.PREVIEW ? (
+        <Label color="blue">{t("preview")}</Label>
+      ) : null}
+      {feature.type === FeatureType.DEFAULT ? (
+        <Label color="green">{t("supported")}</Label>
+      ) : null}
+    </ListItem>
+  );
+};
+
 const Dashboard = () => {
   const { t } = useTranslation("dashboard");
   const { realm } = useRealm();
   const serverInfo = useServerInfo();
   const localeSort = useLocaleSort();
 
-  const isDeprecatedFeature = (feature: string) =>
-    disabledFeatures.includes(feature);
-
-  const isExperimentalFeature = (feature: string) =>
-    serverInfo.profileInfo?.experimentalFeatures?.includes(feature);
-
-  const isPreviewFeature = (feature: string) =>
-    serverInfo.profileInfo?.previewFeatures?.includes(feature);
-
-  const isSupportedFeature = (feature: string) =>
-    !isExperimentalFeature(feature) && !isPreviewFeature(feature);
-
   const disabledFeatures = useMemo(
     () =>
       localeSort(
-        serverInfo.profileInfo?.disabledFeatures ?? [],
-        (item) => item,
+        serverInfo.features?.filter((f) => !f.enabled) || [],
+        mapByKey("name"),
       ),
-    [serverInfo.profileInfo],
+    [localeSort, serverInfo.features],
   );
 
   const enabledFeatures = useMemo(
     () =>
       localeSort(
-        filter(
-          union(
-            serverInfo.profileInfo?.experimentalFeatures,
-            serverInfo.profileInfo?.previewFeatures,
-          ),
-          (feature) => {
-            return !isDeprecatedFeature(feature);
-          },
-        ),
-        (item) => item,
+        serverInfo.features?.filter((f) => f.enabled) || [],
+        mapByKey("name"),
       ),
-    [serverInfo.profileInfo],
+    [localeSort, serverInfo.features],
   );
```

Comment:
I think we can reduce the amount of repetition here by first sorting the features based on the locale, and then using that to split out the enabled and disabled features:

```ts
const sortedFeatures = useMemo(
  () => localeSort(serverInfo.features ?? [], mapByKey("name")),
  [serverInfo.features],
);

const disabledFeatures = useMemo(
  () => sortedFeatures.filter(({ enabled }) => !enabled),
  [sortedFeatures],
);

const enabledFeatures = useMemo(
  () => sortedFeatures.filter(({ enabled }) => enabled),
  [sortedFeatures],
);
```

Also note, that even though the linter complains, we should not add `localeSort` to the dependency array. As this would cause the memo to-retrigger every render.